@jonathan-waarneming-nl er gebeurt wat geks als ik deze update gebruik in de observation repo. Als ik mijn branch bijwerk met deze versie, dus:
```
"@observation.org/react-native-components": "observation/react-native-components#feature/add-disabled-state-to-input-field",
```
in `package.json` zet, dan krijg ik typing errors op `accordion-collapse-react-native`:
```
error TS7016: Could not find a declaration file for module 'accordion-collapse-react-native'.
```
 Met de develop-versie krijg ik dat niet. Maar mijn change verandert daar niets aan. Heb jij enig idee?

EDIT: Gevonden en opgelost. Was een onterechte verwijdering van een `ts-ignore` in de eslint-upgrade.

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@observation.org/react-native-components",
-  "version": "1.44.0",
+  "version": "1.45.0",
   "main": "src/index.ts",
   "repository": "git@github.com:observation/react-native-components.git",
   "author": "Observation.org",

--- a/src/components/InputField.tsx
+++ b/src/components/InputField.tsx
@@ -15,9 +15,19 @@ type Props = {
   rightIcon?: JSX.Element
   description?: string
   errorMessage?: string
+  disabled?: boolean
 }
 
-const InputField = ({ containerStyle, inputProps, inputStyle, label, rightIcon, description, errorMessage }: Props) => {
+const InputField = ({
+  containerStyle,
+  inputProps,
+  inputStyle,
+  label,
+  rightIcon,
+  description,
+  errorMessage,
+  disabled = false,
+}: Props) => {
   const [isFocused, setIsFocused] = useState(false)
 
   // Set lineHeight to 0 to fix vertical alignment of the input text on iOS
@@ -27,6 +37,9 @@ const InputField = ({ containerStyle, inputProps, inputStyle, label, rightIcon, 
   const hasErrors = !!errorMessage
   const borderColor = theme.getBorderColor({ isFocused, hasErrors })
 
+  const inputContainerStyle = disabled ? { backgroundColor: theme.color.greyLight } : {}
+  const placeholderTextColor = disabled ? theme.color.greySemi : theme.color.placeholder
+
   return (
     <View style={[styles.containerStyle, containerStyle]}>
       {label && (
@@ -34,7 +47,7 @@ const InputField = ({ containerStyle, inputProps, inputStyle, label, rightIcon, 
           <Text style={styles.labelStyle}>{label}</Text>
         </View>
       )}
-      <View style={{ flexDirection: 'row' }}>
+      <View style={{ flexDirection: 'row', ...inputContainerStyle }}>
         <TextInput
           style={[{ borderColor }, styles.inputStyle, inputStyle, fixInputStyle]}
           {...inputProps}
@@ -42,7 +55,7 @@ const InputField = ({ containerStyle, inputProps, inputStyle, label, rightIcon, 
           onFocus={() => setIsFocused(true)}
           onBlur={() => setIsFocused(false)}
           underlineColorAndroid="transparent"
-          placeholderTextColor={theme.color.placeholder}
+          placeholderTextColor={placeholderTextColor}
         />
         {rightIcon && <View style={styles.rightIcon}>{rightIcon}</View>}
       </View>

--- a/src/components/__tests__/InputField.test.tsx
+++ b/src/components/__tests__/InputField.test.tsx
@@ -53,6 +53,14 @@ describe('InputField', () => {
       expect(toJSON()).toMatchSnapshot()
     })
 
+    test('Disabled', async () => {
+      // GIVEN
+      const { toJSON } = render(<InputField label="The label" disabled />)
+
+      // THEN
+      expect(toJSON()).toMatchSnapshot()
+    })
+
     test('With an error message', async () => {
       // GIVEN
       const { toJSON } = render(<InputField errorMessage={'The error message'} />)

